framework/src/main/java/org/checkerframework/common/basetype/BaseTypeVisitor.java

```diff
@@ -3103,12 +3103,14 @@ protected void commonAssignmentCheck(
             Tree valueExpTree,
             @CompilerMessageKey String errorKey,
             Object... extraArgs) {
-
-        commonAssignmentCheckStartDiagnostic(varType, valueType, valueExpTree);
-
-        AnnotatedTypeMirror widenedValueType = atypeFactory.getWidenedType(valueType, varType);
-        boolean success = atypeFactory.getTypeHierarchy().isSubtype(widenedValueType, varType);
-
+        boolean success;
+        if (valueExpTree.toString().endsWith(".class")) {
```

Comment:
This is a really bad idea: it converts every Tree to a String and compares it - this will not be good for performance.
It also doesn't assign a proper type to class literals and instead ignores them in comparisons. We should instead look into determining the proper type for class literals.